src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -2035,9 +2035,7 @@ void Cleanup()
             CurrentReader.Command.State = CommandState.Idle;
             try
             {
-                // Will never complete asynchronously (stream is already closed)
-                var readerCloseTask = CurrentReader.CloseAsync();
-                Debug.Assert(readerCloseTask.IsCompleted);
+                CurrentReader.Close();
```

Comment:
Maybe leave a comment just to justify using a sync I/O API in an async path:

```suggestion
                // Note that this never actually blocks on I/O, since the stream is also closed
                // (which is why we don't need to call CloseAsync)
                CurrentReader.Close();
```